datafusion/optimizer/src/unwrap_cast_in_comparison.rs

```diff
@@ -292,6 +296,25 @@ fn is_support_data_type(data_type: &DataType) -> bool {
     )
 }
 
+fn is_decimal_type(dt: &DataType) -> bool {
+    matches!(dt, DataType::Decimal128(_, _))
+}
+
+fn is_unsigned_type(dt: &DataType) -> bool {
+    matches!(
+        dt,
+        DataType::UInt8 | DataType::UInt16 | DataType::UInt32 | DataType::UInt64
+    )
+}
+
+/// Until https://github.com/apache/arrow-rs/issues/1043 is done
+/// (support for unsigned <--> decimal casts) we also don't do that
+/// kind of cast in this optimizer
+fn is_unsupported_cast(dt1: &DataType, dt2: &DataType) -> bool {
```

Comment:
I found this with the test (which failed when it tried to invoke the arrow cast kernels for decimal <--> unsigned)

Reply:
Casting for `unsigned value <-> decimal ` is not supported now.

Reply:
I will try to implement them recently.

Reply:
https://github.com/apache/arrow-rs/pull/3106